package/MDAnalysis/coordinates/PDB.py

```diff
@@ -658,9 +658,11 @@ def __init__(self, filename, bonds="conect", n_atoms=None, start=0, step=1,
         self.first_frame_done = False
 
     def close(self):
-        """Close PDB file and write END record"""
+        """Close PDB file and write CONECT and END record"""
         if hasattr(self, 'pdbfile') and self.pdbfile is not None:
             if not self.has_END:
+                if hasattr(self, "obj"):  # Avoid problems with zero atoms
```

Comment:
Is the no `self.obj` case covered in tests / if not how easy would it be to add a test?

Reply:
Without such safeguard, `test_write_pdb_zero_atoms` was failing.

Reply:
As a thought here, since we're already doing a `self.obj` check in `_write_pdb_bonds()` would it be cleaner / safer (for future changes) to instead have this check there?

Reply:
Probably yes. I'll have a look and change it as you suggest.